solutions/가장긴증가하는부분수열.js

```diff
@@ -0,0 +1,30 @@
+const INPUT_PATH = "../inputs/가장긴증가하는부분수열.txt";
+
+const fs = require("fs");
+const filePath = process.platform === "linux" ? "/dev/stdin" : INPUT_PATH;
+let [sequenceLength, sequence] = fs
+  .readFileSync(filePath)
+  .toString()
+  .trim()
+  .split("\n");
+let result = 0;
+
+sequence = sequence.split(" ").map(Number);
+
+function findAsc(startIndex, currentValue, currentLength) {
+  result = Math.max(result, currentLength);
+
+  for (let i = startIndex; i < Number(sequenceLength); i++) {
+    if (sequence[i] <= currentValue) {
+      continue;
+    }
+
+    findAsc(sequence[i], i, currentLength + 1);
+  }
+}
+
+for (let i = 0; i < sequenceLength; i++) {
+  findAsc(i, sequence[i], 1);
+}
```

Comment:
제 풀이가 최선의 풀이인지는 모르겠으나,,

저는 이진탐색으로 풀어보았습니다.

성호님과 제 코드의 경우 sequence의 각 숫자들에서 for문으로 순회하면서, 빈 결과 배열 값에 길이가 0이거나 마지막 요소가 sequence의 요소보다 작으면 결과배열에 push 한 후, 나머지의 경우에는 이진탐색에서 대치해 주었습니다.

다른 분들의 경우에는 dp 개념으로 이중 for문을 사용하신 것 같은데 해당 풀이도 좋은 것 같습니다!

```suggestion
function solution(input) {
  const sequence = input[1].split(" ").map(Number);
  const resultList = [];

  for (const number of sequence) {
    if (resultList.length === 0 || resultList[resultList.length - 1] < number) {
      resultList.push(number);
    } else {
      let left = 0;
      let right = resultList.length - 1;

      while (left < right) {
        let mid = Math.floor((left + right) / 2);

        if (resultList[mid] < number) {
          left = mid + 1;
        } else {
          right = mid;
        }
      }

      resultList[right] = number;
    }
  }

  return resultList.length;
}
```